IDEOorg/steps-bot
# #164325658 As a developer, I should have faith that in the event the bot crashes, it is automatically restarted

#### What does this PR do?

Adds a package manager library to the api. The library itself is called PM2.

#### Description of Task to be completed?

- Adds PM2 library to the requirements of the bot application.
- The library will restart the application after 3s in the event the application crashes

#### How should this be manually tested?

- clone the repo
- checkout to the branch `ch-integrate-pm2-with-bot-164325658`
- install requirements with `npm install`
- Install `pm2` globally using `npm install --global pm2`.
- Simulate a crash by going to the `index.js` file where the code will execute and add the command `process.exit(1)` on line 25 of the `index.js` file. 
- Run the application using `pm2 start ecosystem.config.js`
- Copy the `ID` field from the result
- Run the command `pm2 logs ID`. `ID` being the value that you copied.
- Notice that the bot crashes and is automatically restarted after every three seconds.
- Stop the process by entering `ctrl` + `c`
- To stop the application, run `pm2 stop ID`. `ID` being the value that you copied.

#### What are the relevant pivotal tracker stories?

[#164325658](https://www.pivotaltracker.com/story/show/#164325658)

#### Screenshots

- N/A

#### Checklist
